DOCS.md

```diff
@@ -0,0 +1,367 @@
+
+// Using evaluate (simple, one-time execution)
+const result1 = evaluate(
+  `export const sum = a + b;`,
+  { a: 1, b: 2 }
+);
+
+// Using compile (more control, reusable)
+const compiled = compile(
+  `export const sum = a + b;`,
+  { a: 1, b: 2 }
+);
+
+const result2 = compiled.run();
+const result3 = compiled.run(); // Can reuse the compiled function
+
+// Both approaches yield the same result
+console.log(result1); // { sum: 3 }
+console.log(result2); // { sum: 3 }
+```
+
+## Error Handling
+
+The `compile` function may throw errors during various phases:
+
+### Parse Errors
+
+```javascript
+import { compile } from "eval-magic";
+
+try {
+  const compiled = compile(`invalid javascript syntax {`);
+} catch (error) {
+  console.error("Parse error:", error.message);
+  // Handle syntax errors in the input code
+}
+```
+
+### Transform Errors
+
+```javascript
+import { compile } from "eval-magic";
+
+// Currently, there's a bug in import function validation
+// This will throw an error even with a valid importFunction
+try {
+  const compiled = compile(
+    `import { unknown } from "missing-module";`,
+    {},
+    {
+      importFunction: (source) => {
+        throw new Error(`Module not found: ${source}`);
+      }
+    }
+  );
+  
+  compiled.run();
+} catch (error) {
+  console.error("Import error:", error.message);
+  // Will show: "import statement is not allowed without an importFunction"
+}
+```
+
+### Configuration Errors
+
+```javascript
+import { compile } from "eval-magic";
+
+try {
+  const compiled = compile(
+    `export const result = x;`,
+    { x: 42 },
+    {
+      parseOptions: {
+        ecmaVersion: 3 // Very old version might cause issues
+      }
+    }
+  );
+} catch (error) {
+  console.error("Configuration error:", error.message);
+}
+```
+
+## Performance Considerations
+
+- **Scope Filtering**: Only variables from `globals` that are actually referenced in the code are included in the execution scope
+- **Reusability**: Compiled functions can be executed multiple times without recompilation
+- **Async Overhead**: Setting `isAsync: true` adds async function wrapper overhead
+
+```javascript
+import { compile } from "eval-magic";
+
+// Compile once, run multiple times for better performance
+const compiled = compile(
+  `export const result = input * 2;`,
+  { input: 0 } // Placeholder, will be filtered out
+);
+
+// This is more efficient than calling evaluate() multiple times
+const results = [1, 2, 3, 4, 5].map(num => {
+  // Note: You'd need to modify the approach for different inputs
+  // This example shows the concept of reusability
+  return compiled.run();
+});
+```
+
+For different inputs, you would typically use parameter passing or recompile with new globals as needed.
+
+## Related Functions
+
+- **[`evaluate`](README.md#usage)**: Higher-level function that compiles and immediately executes code
+- **[`Py` operators](README.md#example)**: Magic method symbols for operator overloading
```

Comment:
Add todos section:

- [ ] allow compiled functions to be called with different `globals` several times without the need to re-compile.